nanoFramework.System.Net.Http/Http/HttpMethod.cs

```diff
@@ -0,0 +1,104 @@
+﻿namespace System.Net.Http
+{
+    /// <summary>
+    /// A helper class for retrieving and comparing standard HTTP methods and for creating new HTTP methods.
+    /// </summary>
+    /// <remarks>
+    /// The most common usage of <see cref="HttpMethod"/> is to use one of the static properties on this class. However, if an app needs a different value for the HTTP method, the <see cref="HttpMethod"/> constructor initializes a new instance of the <see cref="HttpMethod"/> with an HTTP method that the app specifies.
+    /// </remarks>
+    public partial class HttpMethod
+    {
+        private readonly string _method;
+
+        private static readonly HttpMethod s_getMethod = new("GET");
+        private static readonly HttpMethod s_putMethod = new("PUT");
+        private static readonly HttpMethod s_postMethod = new("POST");
+        private static readonly HttpMethod s_deleteMethod = new("DELETE");
+        private static readonly HttpMethod s_headMethod = new("HEAD");
+        private static readonly HttpMethod s_optionsMethod = new("OPTIONS");
+        private static readonly HttpMethod s_patchMethod = new("PATCH");
+
+        /// <summary>
+        /// Represents an HTTP GET protocol method.
+        /// </summary>
+        public static HttpMethod Get
```

Comment:
you can even use the condensed way here  `=> s_getMethod;`
and for the few next ones